pocs/focuser/focuslynx.py

```diff
@@ -299,14 +293,27 @@ def _send_command(self, command_str, expected_reply):
         self._serial_port.reset_input_buffer()
         # Send command
         self._serial_port.write(command_str.encode('ascii'))
-        # Should always get '!' back unless there's an error
         response = str(self._serial_port.readline(), encoding='ascii').strip()
+        if not response:
+            message = "No response to command '{}' from focuser {}".format(command_str, self.uid)
+            self.logger.error(message)
+            raise RuntimeError(message)
+
+        # Should always get '!' back unless there's an error
         if response != '!':
-            message = "Error sending command '{}' to {}: {}".format(command_str, self.uid, response)
+            message = "Error sending command '{}' to focuser {}: {}".format(
+                command_str, self.uid, response)
```

Comment:
Need to add a fallback option to the `uid` getter otherwise this will fail in the case where `_focuser_config` has not been set yet.

Reply:
I got an email with your comment but can't seem to find it here. Add back the check for the `_focuser_config` and show the port otherwise?

Reply:
N/M, I see the comment about the `try/except`. Added in `uid` getter and gives back port if not.

Reply:
I wrote a detailed response to your request for an explanation of the changes, which GitHub almost immediately hid as 'outdated' when you pushed another commit... Its up there, just need to click 'show outdated' until you find it.